web-common/src/features/explores/selectors.ts

```diff
@@ -131,123 +124,32 @@ export async function fetchExploreSpec(
   }
 
   const defaultExplorePreset = getDefaultExplorePreset(
-    exploreResource.explore.state?.validSpec ?? {},
+    exploreSpec,
     fullTimeRange,
   );
+  const { partialExploreState: initExploreState, errors } =
+    convertPresetToExploreState(
+      metricsViewSpec,
+      exploreSpec,
+      defaultExplorePreset,
+    );
+
+  let initLoadedOutsideOfURL = false;
+  const stateFromLocalStorage = restorePersistedDashboardState(
+    exploreSpec,
+    (prefix ?? "") + exploreName,
+  );
+  if (stateFromLocalStorage) {
+    initLoadedOutsideOfURL = true;
+    Object.assign(initExploreState, stateFromLocalStorage);
+  }
 
   return {
     explore: exploreResource,
     metricsView: metricsViewResource,
     defaultExplorePreset,
+    initExploreState,
+    initLoadedOutsideOfURL,
```

Comment:
If we pass the `stateFromLocalStorage` down with the rest of the partial state objects, then I'd hope we don't need to pass down this boolean.